File: blog/3 - Turtle 1.0/F# client/f#/Turtle.fs
//Code provided by Fabio Galuppo 
//April 2012
//A homage to: 
//Logo (http://el.media.mit.edu/logo-foundation/logo/programming.html) and 
//Small Basic(http://msdn.microsoft.com/en-us/beginner/gg605166.aspx)

namespace global

open System

type Turtle() =
   let turtle = new PInvokeTurtle()
   
   member this.close() = (turtle :> IDisposable).Dispose()
       
   member this.rotate(angle: float32) = turtle.rotate(angle)
   
   member this.resize(size: float32) = turtle.resize(size)
   
   member this.move(distance: int) = turtle.move(distance)
   
   member this.speed(value: int) = turtle.speed(value) 

File: blog/3 - Turtle 1.0/Scala client/java/Turtle.java
//Code provided by Fabio Galuppo 
//April 2012
//A homage to: 
//Logo (http://el.media.mit.edu/logo-foundation/logo/programming.html) and 
//Small Basic(http://msdn.microsoft.com/en-us/beginner/gg605166.aspx)

public class Turtle
{
    private JNITurtle turtle;

    public Turtle()
    {
        turtle = new JNITurtle();
    }

    public void close()
    {
        turtle.dispose();
    }

    public void rotate(float angle)
    {
        turtle.rotate(angle);
    }

    public void resize(float size)
    {
        turtle.resize(size);
    }

    public void move(int distance)
    {
        turtle.move(distance);
    }

    public void speed(int value)
    {
        turtle.speed(value);
    }
}